assets/javascripts/discourse-assign/controllers/group-assigned.js.es6

```diff
@@ -12,35 +14,62 @@ export default Controller.extend({
   filterName: "",
   filter: "",
 
+  // I don't know know the subject enough but we should try to remove this one too
+  // one way to do this would be to maybe pass group to group-assigned-filter
+  // and use didReceiveAttrs, I didn't try, but that might be a path forward
+  @observes("model.assignment_count")
+  assignmentCountChanged() {
+    this.set("group.assignment_count", this.model.assignment_count);
+  },
+
   @discourseComputed("site.mobileView")
   isDesktop(mobileView) {
     return !mobileView;
   },
 
-  @observes("filterName")
-  _setFilter: discourseDebounce(function() {
-    this.set("filter", this.filterName);
-  }, 500),
+  @action
+  loadMore() {
+    this.findMembers();
+  },
 
-  @observes("filter")
-  _filterModel() {
+  // we try to remove observers as much as possible, also look at assigned.hbs to see what I did here
+  @action
+  onChangeFilterName(value) {
+    // we have a global for INPUT_DELAY, feel free to use more, you could also do
+    // INPUT_DELAY * 2 if you feel that's necessary
+    // note that any debounce couldn't end up happening after the lifecycle of your object
+    // eg do something with x in 2 seconds..... x.doSomething()... problem is... x doesn't exist anymore
+    // this might be ok here, but it's something to keep in mind
+    debounce(this, this._setFilter, value, INPUT_DELAY);
+  },
+
+  _setFilter(filter) {
     this.set("loading", true);
     this.set("offset", 0);
-    ajax(`/assign/members/${this.group.name}`, {
+    this.set("filter", filter);
+
+    const groupName = this.group.name;
+
+    // I generally try to always return the prommise if there's one
+    return ajax(`/assign/members/${groupName}`, {
       type: "GET",
       data: { filter: this.filter, offset: this.offset }
-    }).then(result => {
-      if (this.router.currentRoute.params.filter !== "everyone") {
-        this.transitionToRoute("group.assigned.show", "everyone");
-      }
-      this.set("members", result.members);
-      this.set("loading", false);
-    });
-  },
+    })
+      .then(result => {
+        // know that this is risky, and a bug actually in your code
+        // if you type something in filter and click fast on logo befire the end of debouncing
+        // you will load home and then reload group.assigned.show
+        if (this.router.currentRoute.params.filter !== "everyone") {
```

Comment:
@jjaffeux how do you feel about accessing the route service like this in a controller? Ahmed and I have been talking in [this PR](https://github.com/discourse/discourse-assign/pull/98#discussion_r470098636) about it... 

Originally I was thinking it should be avoided... but now I'm reading [the ember docs](https://api.emberjs.com/ember/3.9/classes/RouterService/methods) about the service and it does seem to be designed for use by controllers and components. So maybe this is fine? 🤷‍♂️

Reply:
I would yes it's fine, and you are only accessibg public things so 🤷‍♂️